File: scratch/restore.py
import json
import re
import sys

transcript_path = r"C:\Users\MFM IKOT ABASI\.gemini\antigravity\brain\6e170f61-89b2-4ce9-bc72-09afd550ce5d\.system_generated\logs\transcript.jsonl"
out_path = r"C:\Users\MFM IKOT ABASI\Documents\nd\admin\request\request.js"

original_content = None

with open(transcript_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            data = json.loads(line)
            if data.get('type') == 'TOOL_RESPONSE' and 'request.js' in data.get('content', ''):
                content = data['content']
                # Search for the view_file output block for request.js
                if 'File Path: `file:///c:/Users/MFM%20IKOT%20ABASI/Documents/nd/admin/request/request.js`' in content:
                    original_content = content
                    break
        except Exception as e:
            pass

if original_content:
    # Extract the lines of code. It starts after "The following code has been modified..."
    lines = original_content.split('\n')
    extracted_lines = []
    in_code = False
    for line in lines:
        if line.startswith('1: '):
            in_code = True
        
        if in_code:
            if re.match(r'^\d+: ', line):
                # Remove the line number
                clean_line = line.split(': ', 1)[1]
                extracted_lines.append(clean_line)
            elif line.startswith('The above content'):
                in_code = False
                break
            else:
                extracted_lines.append(line)
                
    # write to file
    with open(out_path, 'w', encoding='utf-8') as f:
        f.write('\n'.join(extracted_lines))
        
    print(f"Successfully restored {len(extracted_lines)} lines to request.js")
else:
    print("Could not find the original request.js content in transcript.")

File: scratch/restore.js
const fs = require('fs');

const transcript_path = "C:\\Users\\MFM IKOT ABASI\\.gemini\\antigravity\\brain\\6e170f61-89b2-4ce9-bc72-09afd550ce5d\\.system_generated\\logs\\transcript.jsonl";
const out_path = "C:\\Users\\MFM IKOT ABASI\\Documents\\nd\\admin\\request\\request.js";

const lines = fs.readFileSync(transcript_path, 'utf-8').split('\n');
let original_content = null;

for (const line of lines) {
    if (!line.trim()) continue;
    try {
        const data = JSON.parse(line);
        if (data.type === 'TOOL_RESPONSE' && data.content) {
            if (data.content.includes('Total Lines: 822') && data.content.includes('let currentRequestFilter')) {
                original_content = data.content;
                break;
            }
        }
    } catch (e) {}
}

if (original_content) {
    const textLines = original_content.split('\n');
    const extracted_lines = [];
    let in_code = false;
    for (const line of textLines) {
        if (line.startsWith('1: ')) {
            in_code = true;
        }
        if (in_code) {
            const match = line.match(/^\d+: (.*)/);
            if (match) {
                extracted_lines.push(match[1]);
            } else if (line.startsWith('The above content') || line.startsWith('The above content does NOT show')) {
                in_code = false;
                break;
            } else {
                extracted_lines.push(line);
            }
        }
    }
    fs.writeFileSync(out_path, extracted_lines.join('\n'), 'utf-8');
    console.log("Successfully restored " + extracted_lines.length + " lines");
} else {
    console.log("Not found.");
}
